tests/bdd/features/snapshot/create/feature.feature

```diff
@@ -0,0 +1,78 @@
+Feature: Volume Snapshot Creation
+
+  Background:
+    Given a deployer cluster
+
+  Scenario Outline: Snapshot creation of a single replica volume
+    Given we have a single replica <publish_status> volume where the replica is <replica_location> to the target
+    When we try to create a snapshot for the volume
+    Then the snapshot creation should be successful
+    And the snapshot status ready to clone should be true
+    And the snapshot source size should be equal to the volume size
+    And the snapshot size should be equal to the volume size
+    And the snapshot timestamp should be in the RFC 3339 format
+    And the snapshot timestamp should be within 1 minute of creation (UTC)
+    And the volume snapshot state has a single online replica snapshot
+    And the replica snapshot's timestamp should be in the RFC 3339 format
+    And the replica snapshot's timestamp should be within 1 minute of creation (UTC)
+    Examples:
+      | publish_status | replica_location |
+      | published      | remote           |
+      | published      | local            |
+      | unpublished    | remote           |
+      | unpublished    | local            |
+
+  Scenario: Snapshot creation for a single replica volume fails when the replica is offline
+    Given we have a single replica volume
+    And the replica is offline
+    And volume is in faulted state
+    When we try to create a snapshot for the volume
+    Then the snapshot creation should fail with preconditions failed
+    And the snapshot should not be listed
+
+  Scenario: Snapshot creation for multi-replica volume is not supported
+    Given we have a multi-replica volume
+    When we try to create a snapshot for the volume
+    Then the snapshot creation should be fail with preconditions failed
+
+  Scenario: Existence of snapshot after node restart
+    Given we have a single replica volume
+    And a successfully created snapshot
+    When we stop the node hosting the snapshot
```

Comment:
Instead of stopping and starting can we just restart?

Reply:
And if we want to stop and start then probably should we kill and start, i.e ungracefully.

Reply:
why does it matter whether we stop or kill here?

Reply:
Stop does things gracefully, which might not be the case always.

Reply:
Vice versa :) upgrades etc will use stop

Reply:
Maybe we could use outline and test both

Reply:
Right.